corehq/util/metrics/__init__.py

```diff
@@ -267,6 +269,8 @@ def stop(self, name=None):
             self._metric, self.duration, bucket_tag='duration',
             buckets=self._timing_buckets, bucket_unit='s', tags=self._tags
         )
+        if self._errored:
+            metrics_counter(self._metric, tags={**self._tags, 'duration': 'error'})
```

Comment:
For datadog this means it'll receive histogram metrics like this:

- commcare.metric duration:lt_001s
- commcare.metric duration:lt_010s
- commcare.metric duration:over_010s
- commcare.metric duration:error

I'm less familiar with prometheus - the histogram implementation there is done in their library, and it the values passed in must all be integers (understandably).  This will result in sending a histogram metric _and_ a counter metric with the same name.  I'm not sure how prometheus will handle this, nor how to test it.  I suppose an alternative proposal would be to do something like:
```suggestion
            metrics_counter(f'{self._metric}.error', tags={**self._tags})
```
Thoughts?